Feature/next/article

dw-buildit opened the pull request:
This PR is two components and some global style updates.

The two components are:

**Page Intro**
To be used as the `<heading>` for most of our pages. It includes the heading, description and 
optional time and image as per reference designs.

**Ostentatious Copy**
This is expected to contain copy that you want to look a little more showy. Generally adds extra padding between elements and around the component itself. Pulls pull quotes out (margin wise) and on larger screen sizes indents a lot from the left.

**Style Updates**
The PR also includes updates to many text element, list element and some padding styles. Again to better reflect the reference design.

**Also**
There's styling for the `<hr>` element.
There's a page for viewing all the article components with semi-realistic data.

ashblue commented:
For the 2nd commit `feat(notch decoration): added parameters to the mixin` is that really a feature? I feel like it should fall under a chore or refactor because it isn't declaring anything new for the repository.

james-nash commented:
Just adding a cross reference to issue #178, since this partially covers that.

Partially, since we don't yet have provisions for article author, tags and/or categories. However, all of those could be added in a future PR, after the website MVP is complete.

dw-buildit commented:
> For the 2nd commit `feat(notch decoration): added parameters to the mixin` is that really a feature? I feel like it should fall under a chore or refactor because it isn't declaring anything new for the repository.

I did think that this was a feature that developers of Gravity would be able to use (dynamic notch size/position) although I wouldn't actually expect any users to use it. I've changed this commit's message anyway 👍

james-nash commented on packages/gravity-ui-nunjucks/src/_patterns/03-organisms/00-page-structure/04-page-intro.md:
```diff
@@ -0,0 +1,5 @@
+
+This is the component for use as the page heading.
+
+It consists of a `<h1>` as the page title, a `<p>` for a short description of the page, an optional `<time>` to display a date related to this content (usually publication date), and an optional `<img>` which will automatically gain a downward pointing notch.
```
Maybe add some guidance here about when to use a `<header>` element or a `<div>` to markup the page intro.

james-nash commented on packages/gravity-ui-nunjucks/src/_patterns/05-pages/05-article.njk:
```diff
@@ -0,0 +1,29 @@
+  </article>
+  <hr />
+  <aside class="grav-o-container">
+    <ul class="grav-c-related-items">
```
You should replace this with `{% include 'organisms-related-items' %}`. That way, if the related items component ever changes, it will be reflected here. Also, it lets Pattern Lab / Fractal create lineage links between those components in the pattern library.

dw-buildit replied:
Definitely! That was an oversight.

james-nash commented on packages/gravity-ui-web/src/sass/05-components/03-organisms/_copy-ostentatious.scss:
```diff
@@ -0,0 +1,38 @@
+  }
+
+  @media all and (max-width: grav-breakpoint(medium)) {
+    padding-right: $grav-sp-s;
```
We can avoid this `max-width` media query. You can set this padding and the blockquote styles outside of the media query.

Then, below, in the `min-width` media query, where you already alter the blockquote styles as needed, you can update the padding to be large instead of small.

dw-buildit commented:
@james-nash CR amends completed.

buildit-dev commented:
:tada: This PR is included in version 3.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@buildit/gravity-ui-web/v/3.0.0)
- [GitHub release](https://github.com/buildit/gravity-ui-web/releases/tag/gravity-ui-web-v3.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
